### PR TITLE
fix: only reset syncController if we passed a discontinuity

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by gkatsev** [comment]
Unless anyone gets a good idea, I don't think there's a good test to add for this.

**Comment by gkatsev** [comment]
Doesn't seem to cause issues with demuxed audio stream either.

File: test/segment-loader.test.js [hunk 1]
```diff
@@ -442,6 +442,42 @@ QUnit.module('SegmentLoader: M2TS', function(hooks) {
...
+      this.requests.shift().respond(200, null, '');
+      buffered = videojs.createTimeRanges([[0, 10]]);
+      this.updateend();
+      this.clock.tick(1);
+
+      assert.equal(hlsTimestampOffsetEvents, 0,
+        'no hls-timestamp-offset event was fired');
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+      loader.sourceUpdater_.timestampOffset_ = 11;
```
Comment (mjneil) [inline test/segment-loader.test.js]: what is this line doing?

Comment (gkatsev) [reply test/segment-loader.test.js]: terrible things but it's responsible for making the conditional that was removed pass

Comment (mjneil) [reply test/segment-loader.test.js]: I think you would want to do this before the updateend or clock tick above 465-466

File: src/segment-loader.js [hunk 2]
```diff
@@ -742,9 +742,7 @@ export default class SegmentLoader extends videojs.EventTarget {
     // - The "timestampOffset" for the start of this segment is less than
```
Comment (mjneil) [inline src/segment-loader.js]: Comment should be updated to reflect changes

File: test/segment-loader.test.js (hunk 1 above)
Comment (gkatsev) [reply test/segment-loader.test.js]: apparently, nothing.

**Comment by gkatsev** [comment]
Can't come up with a better test, unfortunately.

**Pull request merged**